### PR TITLE
Add signin functionality with JWT

### PR DESCRIPTION
**Pull request by emanuelcasco**
## Summary

- Add "invalid user" error to handle bad authentication attempts.
- Add "Signin" test.
  - Also fixed issue in data creation script.
  - This PR also updates docs (dictum).
- Add signin data validation.
  - Also made little refactor. 
- Add authentication middleware. It is not used yet, but is part of sign in functionality and upcoming PR (admin user) will need it.
- Add session manager to handle authentication. It only purpose is encode and decode data using JWT.

## Known Issues

None

## Trello Card

- https://trello.com/c/tVFTYJVl/3-sign-in

### CODE REVIEW

File: test/users.test.js [hunk 1]
```diff
@@ -135,4 +136,89 @@ describe('users', () => {
...
+          password: 'password1234',
+          email: 'joe.doe@wolox.com.ar'
+        })
+        .then(res => {
+          res.should.have.status(200);
+          res.should.be.json;
+          res.headers.should.have.property(sessionManager.HEADER_NAME);
+          sessionManager
+            .decode(res.headers[sessionManager.HEADER_NAME])
+            .should.be.equal('joe.doe@wolox.com.ar');
```
Comment (darkBasic666) [inline test/users.test.js]: why do you checl sessionManager inside of succes response ?

File: test/users.test.js [hunk 2]
```diff
@@ -135,4 +136,89 @@ describe('users', () => {
...
+        .request(server)
+        .post('/users/sessions')
+        .send({
+          password: 'password1234',
+          email: 'joe.doe@other.com'
+        })
+        .then(res => {
+          res.should.have.status(400);
+          res.should.be.json;
+          res.body.should.have.property('message');
```
Comment (darkBasic666) [inline test/users.test.js]: why do you don't check message text?

File: test/users.test.js [hunk 3]
```diff
@@ -135,4 +136,89 @@ describe('users', () => {
...
+        .request(server)
+        .post('/users/sessions')
+        .send({
+          password: 'short12',
+          email: 'joe.doe@wolox.com.ar'
+        })
+        .then(res => {
+          res.should.have.status(400);
+          res.should.be.json;
+          res.body.should.have.property('message');
```
Comment (darkBasic666) [inline test/users.test.js]: same

File: test/users.test.js [hunk 4]
```diff
@@ -135,4 +136,89 @@ describe('users', () => {
...
+        .request(server)
+        .post('/users/sessions')
+        .send({
+          password: 'password1234',
+          email: 'not.exist@wolox.com.ar'
+        })
+        .then(res => {
+          res.should.have.status(400);
+          res.should.be.json;
+          res.body.should.have.property('message');
```
Comment (darkBasic666) [inline test/users.test.js]: same

File: test/users.test.js [hunk 5]
```diff
@@ -135,4 +136,89 @@ describe('users', () => {
...
+        .request(server)
+        .post('/users/sessions')
+        .send({
+          password: '123notcorrect',
+          email: 'joe.doe@wolox.com.ar'
+        })
+        .then(res => {
+          res.should.have.status(400);
+          res.should.be.json;
+          res.body.should.have.property('message');
```
Comment (darkBasic666) [inline test/users.test.js]: same

File: test/users.test.js (hunk 2 above)
Comment (emanuelcasco) [reply test/users.test.js]: I forget error messages, my bad. Resolved!

File: test/users.test.js (hunk 1 above)
Comment (emanuelcasco) [reply test/users.test.js]: I want to test if user is correctly authenticated. So, I verify 'authorization' property in header was defined and it correspond with my user's e-mail

**Pull request merged**